### PR TITLE
config: Enable config for suppressing update-notifier

### PR DESCRIPTION
**Pull request by travi**
based on conversation from https://npm.community/t/npm-side-config-for-suppressing-update-notifier/75

### CODE REVIEW

**Comment by travi** [comment]
i'm happy to make any adjustments necessary, but this should be close. thanks for being open to the contribution

**Review by zkat — APPROVED** [review]
This looks pretty much about right! Thank you!

File: bin/npm-cli.js [hunk 1]
```diff
@@ -78,6 +78,7 @@
       const pkg = require('../package.json')
       let notifier = require('update-notifier')({pkg})
       if (
+        npm.config.get('update-notifier'),
```
Comment (zkat) [inline bin/npm-cli.js]: Actually, I'd like this to be further up so you don't even require the notifier package at all?

**Comment by travi** [comment]
looks like i missed a simple syntax issue that `npm t` would have pointed out to me had i run it before pushing :\

ill have a fix up in a min

File: bin/npm-cli.js (hunk 1 above)
Comment (travi) [reply bin/npm-cli.js]: sure, that makes sense. how about combined w/ the conditional on line 77?

**Comment by travi** [comment]
i'm having some trouble getting the tests to run correctly locally, but the travis output gives me proper insight with a longer feedback loop. i'll wrap those up tomorrow.

**Comment by travi** [comment]
ah, it looks like [a few tests](https://github.com/npm/npm/blob/4c65cd952bc8627811735bea76b9b110cc4fc80e/test/tap/config-meta.js#L111-L131) that ensure the new config var is used somewhere in the code, but it [doesnt seem to check files under `bin/`](https://github.com/npm/npm/blob/4c65cd952bc8627811735bea76b9b110cc4fc80e/test/tap/config-meta.js#L38-L52). do you want me to update the file search to include `bin/` in addition to `lib/`?

**Comment by zkat** [comment]
Gonna cc @iarna on that q (I'm off work today)

**Comment by travi** [comment]
sure thing. i'm happy to make the updates if thats best, but want to focus on what solution you see as most valuable. 

enjoy the day off @zkat

**Comment by travi** [comment]
i went ahead and included `bin/` in the search for config usages and it does get the tests passing again. i made that change as a separate commit so i can easily back that out if you want to take a different approach. if you want to move forward with this approach, i can squash the two together, if you'd like.

**Comment by travi** [comment]
anything else needed on my side to move this forward?

**Comment by zkat** [comment]
Everything looks good :)

**Comment by travi** [comment]
thanks!

**Pull request merged**